PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/localrun/FunctionRunConfiguration.java

```diff
@@ -0,0 +1,176 @@
+        return new FunctionRunSettingEditor(getProject(), this);
+    }
+
+    @Nullable
+    @Override
+    public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment executionEnvironment) throws ExecutionException {
+        return new FunctionRunState(getProject(), this, executor);
+    }
+
+    public Module getModule() {
+        return module == null ? FunctionUtils.getFunctionModuleByFilePath(getProject(), getModuleFilePath()) : module;
+    }
+
+    public JsonObject getAppSettingsJsonObject() {
+        return appSettingsJsonObject;
+    }
+
+    public String getDebugOptions() {
+        return functionRunModel.getDebugOptions();
+    }
+
+    public void setDebugOptions(String debugOptions) {
+        functionRunModel.setDebugOptions(debugOptions);
+    }
+
+    public String getStagingFolder() {
+        return functionRunModel.getStagingFolder();
+    }
+
+    public String getFuncPath() {
+        return functionRunModel.getFuncPath();
+    }
+
+    public String getHostJsonPath() {
+        return functionRunModel.getHostJsonPath();
+    }
+
+    public String getLocalSettingsJsonPath() {
+        final String path = functionRunModel.getLocalSettingsJsonPath();
+        return StringUtils.isNotEmpty(path) ? path : Paths.get(getProject().getBasePath(), "local.settings.json").toString();
+    }
+
+    public Map<String, String> getAppSettings() {
+        return functionRunModel.getAppSettings();
+    }
+
+    public String getModuleFilePath() {
+        return functionRunModel.getModuleFilePath();
+    }
+
+    public FunctionRunModel getFunctionRunModel() {
+        return functionRunModel;
+    }
+
+    public void setLocalSettingsJsonPath(String localSettingsJsonPath) {
+        functionRunModel.setLocalSettingsJsonPath(localSettingsJsonPath);
+    }
+
+    public void setAppSettings(Map<String, String> appSettings) {
+        functionRunModel.setAppSettings(appSettings);
+    }
+
+    public void setModuleFilePath(String moduleName) {
+        functionRunModel.setModuleFilePath(moduleName);
+    }
+
+    public void setFunctionRunModel(FunctionRunModel functionRunModel) {
+        this.functionRunModel = functionRunModel;
+    }
+
+    public void setAppSettingsJsonObject(JsonObject appSettingsJsonObject) {
+        this.appSettingsJsonObject = appSettingsJsonObject;
+    }
+
+    public void setModule(Module module) {
+        this.module = module;
+        this.setModuleFilePath(module.getModuleFilePath());
+    }
+
+    public void setStagingFolder(String stagingFolder) {
+        functionRunModel.setStagingFolder(stagingFolder);
+    }
+
+    public void setFuncPath(String funcPath) {
+        functionRunModel.setFuncPath(funcPath);
+    }
+
+    public void setHostJsonPath(String hostJsonPath) {
+        functionRunModel.setHostJsonPath(hostJsonPath);
+    }
+
+    @Override
+    public void validate() throws ConfigurationException {
```

Comment:
shall here validate func.exe, staging folder(valid directory name), host json existence?

Reply:
Will add the validation in later PR